File: python/copass-management/src/copass_management/tools/grant_sandbox_connection.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def grant_sandbox_connection(
    ctx: "ToolContext", input: Dict[str, Any],
) -> Any:
    kwargs: Dict[str, Any] = {"role": str(input["role"])}
    if input.get("copass_id") is not None:
        kwargs["copass_id"] = str(input["copass_id"])
    if input.get("user_id") is not None:
        kwargs["user_id"] = str(input["user_id"])
    if input.get("project_id") is not None:
        kwargs["project_id"] = str(input["project_id"])
    if input.get("label") is not None:
        kwargs["label"] = str(input["label"])
    if input.get("expires_at") is not None:
        kwargs["expires_at"] = str(input["expires_at"])
    return await ctx.client.sandbox_connections.create(
        ctx.sandbox_id, **kwargs,
    )

File: typescript/packages/management/src/tools/grant_sandbox_connection.ts
import type {
  ConnectionRole,
  CreateSandboxConnectionRequest,
} from '@copass/core';

import type { ToolContext, ToolHandler } from '../registrar.js';

export const grantSandboxConnection: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const role = String(input.role) as ConnectionRole;
  const request: CreateSandboxConnectionRequest = { role };
  if (typeof input.copass_id === 'string') {
    request.copass_id = input.copass_id;
  }
  if (typeof input.user_id === 'string') {
    request.user_id = input.user_id;
  }
  if (typeof input.project_id === 'string') {
    request.project_id = input.project_id;
  }
  if (typeof input.label === 'string') {
    request.label = input.label;
  }
  if (typeof input.expires_at === 'string') {
    request.expires_at = input.expires_at;
  }
  return ctx.client.sandboxConnections.create(ctx.sandboxId, request);
};
